syntax = "proto3";

package mes_grpc;

option go_package = "proto/;mes_grpc";

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

service MessengerServer {
  rpc SendMessage(SendRequest) returns (SendResponse);
  rpc ReadMessages(google.protobuf.Empty) returns (stream ReadResponse);
}

message SendRequest {
  string author = 1;
  string text = 2;
}

message SendResponse {
    google.protobuf.Timestamp sendTime = 1;
}

message ReadResponse {
    string author = 1;
    string text = 2;
    google.protobuf.Timestamp sendTime = 3;
}

// TODO: Add messages and service
// service must contain the following rpc methods: SendMessage and ReadMessages